client/core/core.go

```diff
@@ -1272,18 +1273,30 @@ func (c *Core) Run(ctx context.Context) {
 		}
 		symb := strings.ToUpper(unbip(assetID))
 		c.log.Infof("Locking %s wallet", symb)
-		if err := wallet.Lock(); err != nil {
+		if err := walletLockWithTimeout(5*time.Second, wallet); err != nil {
 			c.log.Errorf("Failed to lock %v wallet: %v", symb, err)
 		}
 		wallet.Disconnect()
 	}
 
-	// Let the wallet backends shut down cleanly.
-	c.walletWait.Wait()
-
 	c.log.Infof("DEX client core off")
 }
 
+func walletLockWithTimeout(timeout time.Duration, wallet *xcWallet) error {
```

Comment:
It looks like you can just add a timeout argument to `(*xcWallet).Lock` and move this there, since `walletLockWithTimeout` is the only place it's used now.

Reply:
Gave this a thumbs up, but it's looking really quite smart to move it like you said because inside `(*xcWallet).Lock` has `xcWallet.mtx` is locked while it sits on `asset.Wallet.Lock`  Looks like we'd have ended up with deadlocks anyway if there were other `xcWallet` method calls.
Going to try to ensure the mutex is not locked while it's waiting on the backend.